src/items.rs

```diff
@@ -1267,6 +1267,25 @@ pub fn rewrite_associated_type(ident: ast::Ident,
     }
 }
 
+pub fn rewrite_associated_impl_type(ident: ast::Ident,
+                                    defaultness: ast::Defaultness,
+                                    ty_opt: Option<&ptr::P<ast::Ty>>,
+                                    ty_param_bounds_opt: Option<&ast::TyParamBounds>,
+                                    context: &RewriteContext,
+                                    indent: Indent)
+                                    -> Option<String> {
+    let result = rewrite_associated_type(ident, ty_opt, ty_param_bounds_opt, context, indent);
+
+    if let ast::Defaultness::Default = defaultness {
+        match result {
```

Comment:
you could skip this match block by wrapping the rhs of `let result = ...` in `try_opt!`